From: Michal Simek <michal.simek@amd.com>
Date: Mon, 15 Jul 2024 08:24:57 +0200
Subject: [PATCH] arm64: versal: Fix vek280 memory configuration

Board has 12GB total but Vivado wizard supports one channel LPDDR4 4GB
configuration (2GB low + 2GB high) or two channel LPDDR 8GB configuration
(2GB low + 6GB high). Wiring the remaining 4GB LPDDR can be done via board
tab but not available as option in wizard. That's why only describe 8GB by
default.

Signed-off-by: Michal Simek <michal.simek@amd.com>
Signed-off-by: Neal Frager <neal.frager@amd.com>
Upstream: https://github.com/Xilinx/linux-xlnx/commit/3c7a97d99bdb3dd6b1f36d7d87d3a3c113dfe2ea
---
 arch/arm64/boot/dts/xilinx/versal-vek280-revA.dts | 2 +-
 arch/arm64/boot/dts/xilinx/versal-vek280-revB.dts | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/xilinx/versal-vek280-revA.dts b/arch/arm64/boot/dts/xilinx/versal-vek280-revA.dts
index 1641dc0b5141..ad5d590318e6 100644
--- a/arch/arm64/boot/dts/xilinx/versal-vek280-revA.dts
+++ b/arch/arm64/boot/dts/xilinx/versal-vek280-revA.dts
@@ -19,7 +19,7 @@ / {
 
 	memory: memory@0 {
 		device_type = "memory";
-		reg = <0 0 0 0x80000000>, <0x8 0x0 0x7 0x80000000>; /* 32GB */
+		reg = <0 0 0 0x80000000>, <0x8 0x80000000 0x1 0x80000000>; /* 2GB + 6GB */
 	};
 
 	chosen {
diff --git a/arch/arm64/boot/dts/xilinx/versal-vek280-revB.dts b/arch/arm64/boot/dts/xilinx/versal-vek280-revB.dts
index fda47cd87b3a..80aae8fa4b11 100644
--- a/arch/arm64/boot/dts/xilinx/versal-vek280-revB.dts
+++ b/arch/arm64/boot/dts/xilinx/versal-vek280-revB.dts
@@ -19,7 +19,7 @@ / {
 
 	memory: memory@0 {
 		device_type = "memory";
-		reg = <0 0 0 0x80000000>, <0x8 0x0 0x7 0x80000000>; /* 32GB */
+		reg = <0 0 0 0x80000000>, <0x8 0x80000000 0x1 0x80000000>; /* 2GB + 6GB */
 	};
 
 	chosen {
-- 
2.43.0

